quixstreams/app.py

```diff
@@ -920,23 +927,28 @@ def _on_revoke(self, _, topic_partitions: List[TopicPartition]):
             self._processing_context.commit_checkpoint(force=True)
 
         self._consumer.incremental_unassign(topic_partitions)
-        if self._state_manager.stores:
-            logger.debug(f"Rebalancing: revoking state store partitions")
-            for tp in topic_partitions:
+        for tp in topic_partitions:
+            if self._state_manager.stores:
                 self._state_manager.on_partition_revoke(
                     topic=tp.topic, partition=tp.partition
                 )
+            self._processing_context.on_partition_revoke(
+                topic=tp.topic, partition=tp.partition
+            )
 
     def _on_lost(self, _, topic_partitions: List[TopicPartition]):
         """
         Dropping lost partitions from consumer and state
         """
+        logger.debug(f"Rebalancing: dropping lost partitions")
         if self._state_manager.stores:
-            logger.debug(f"Rebalancing: dropping lost state store partitions")
             for tp in topic_partitions:
                 self._state_manager.on_partition_revoke(
                     topic=tp.topic, partition=tp.partition
                 )
+                self._processing_context.on_partition_revoke(
```

Comment:
This won't be trigger if `self._state_manager.stores` is falsy

Reply:
Yep, I know. 
I moved these debug logs to the top in the other handlers too.

Reply:
Yep, forgot to update `on_lost` too